src/main/java/uk/gov/justice/hmpps/prison/service/receiveandtransfer/BookingIntoPrisonService.kt

```diff
@@ -73,13 +72,14 @@ class BookingIntoPrisonService(
   authenticationFacade = authenticationFacade,
 ) {
   fun newBooking(prisonerIdentifier: String, requestForNewBooking: RequestForNewBooking): InmateDetail {
-    return newBooking(offender(prisonerIdentifier).getOrThrow(), requestForNewBooking)
+    // grab a lock on the offender and their bookings to ensure that no-one else can add a booking to the offender
+    // during our transaction.  Other processes / threads will hang waiting for this to complete.
+    val offender = offenderForUpdate(prisonerIdentifier).getOrThrow()
+    return newBookingWithoutUpdateLock(offender, requestForNewBooking)
   }
-  fun newBooking(offender: Offender, requestForNewBooking: RequestForNewBooking): InmateDetail {
-    // ensure that we can get a select for update on the bookings before we start
-    val bookings = offenderBookingRepository.findAllByOffenderNomsIdForUpdate(offender.nomsId)
```

Comment:
by grabbing the bookings separately we ended up with two different collections of bookings - one on the offender and then one here.  This meant that when hibernate the saved it saved both and one ended up overwriting the other one.  By switching to using and locking the offender we only end up with one list of bookings